backend/src/features/template/template.service.ts

```diff
@@ -0,0 +1,91 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { PrismaService } from '@prisma/prisma.service';
+import {
+  TemplateBodyRequestDto,
+  TemplateIdRequestDto,
+  TemplateQuery,
+  TemplateRequestDto,
+  TemplatesResponseDto,
+} from '@shared/models/template';
+import { ErrorText } from '@shared/utils/enums';
+import { parseBaseQuery } from '@shared/utils/funcs/parse-base-query.func';
+
+@Injectable()
```

Comment:
What is purpose of injectable?

Reply:
this question cant be answered fully in a one comment here xd
but basically an `@Injectable()` is a class which takes on purpose of a Service or Provider in NEST.js app. Service or Provideres are classes which provide additional functionality, which should be seperate from the functionality of a Controller (which is to provide API endpoints control).